### PR TITLE
JS-1597 add generated-source foundation

### PR DESCRIPTION
**Pull request by francois-mora-sonarsource**
Part of JS-1597

## What
- introduce the generated-source file store and shared detector/task-invocation contracts
- derive generated-source metadata from dependency manifests and normalized package.json task invocations
- wire generated-source awareness into analysis configuration and file-store initialization
- add project-metadata tests for invocation normalization, empty-derivation behavior, and store behavior with no registered detector

## Why
- generated-source detection needs common plumbing before tool-specific detectors can be added safely
- this change establishes the shared APIs, watched-path handling, and store lifecycle for the follow-up PRs in this series

## Impact
- no detector is registered yet, so behavior should remain unchanged for analyzed projects
- follow-up changes can add detectors on top of the same derived metadata and store contract

## Validation
- `npm run bridge:build:fast`
- `npx tsx --tsconfig packages/tsconfig.test.json --test --test-reporter=spec --test-reporter-destination stdout packages/analysis/tests/jsts/project-metadata/generated-sources.test.ts`

----
## Summary by Gitar

- **Refactored file store lifecycle:**
  - Updated `FileStore` interface to support `postProcess` with optional `analyzableFiles`.
  - Added configuration hashing via `getProjectFileDiscoveryConfigKey` and `getAnalyzableFilesConfigKey` for granular cache invalidation across all stores.
- **New generated-source store:**
  - Added `generatedSourceStore` to coordinate detection, watched paths, and family mappings.
  - Integrated generated-source detection into `analyzeProject` via `isGeneratedSourceFile` to classify files for the linter.
- **Detector foundation:**
  - Implemented `GeneratedSourceDetector` contract and `deriveGeneratedSources` to collect metadata from package manifests and task invocations.

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

**Comment by francois-mora-sonarsource** [comment]
Apart from the new generated-source store, this PR tightens the cache contract of the existing file stores.

Before this change, the existing stores were reused mostly based on `baseDir` plus a few store-specific invalidation signals. That was too coarse. A store can become stale even when the project root stays the same if:
- the set of analyzable files changes
- the set of discoverable helper files changes

To make that explicit, `configuration.ts` now exposes two cache domains:
- `analyzable-file selection`: inputs that decide which source files belong to the analysis
- `project-helper discovery`: inputs that decide which helper files can be discovered during the project walk

The changes to the pre-existing stores are:

- `sourceFileStore` now has clearer refresh rules for its two existing usage patterns.
  In request-driven analysis, where `inputFiles` are provided, the behavior is unchanged in principle: the store is still rebuilt for every request, so SonarQube for IDE does **not** start reusing the previous request’s source-file set. The fix here is that the rebuild now starts from a clean state. That prevents stale `ignoredPaths` and stale `DirectoryIndex` entries from leaking from one request to the next.
  In filesystem-driven analysis, where files are discovered by walking `baseDir`, reuse no longer depends only on `baseDir`: it now also depends on the analyzable-file-selection inputs (`sources`, `tests`, inclusions/exclusions, suffixes, bundle detection, max file size, `jsTsExclusions`, and related filters).
  Why: the source-file set can change without changing the project root, and request-driven rebuilds should not accumulate state from previous requests.

- `dependencyManifestStore` now invalidates on the project-helper-discovery cache key, in addition to manifest-related file events.
  Why: dependency manifests are discovered through the project walk. If helper-file discovery conditions change, the set of manifests that can be found may also change, even when `baseDir` does not.

- `tsConfigStore` now follows the same invalidation model for discovered tsconfigs, while still keeping `tsConfigPaths`, `clearTsConfigCache`, and tsconfig-related file events as its own refresh triggers.
  Why: discovered `tsconfig.json` files depend on helper-file discovery conditions, not on which subset of source files is currently being analyzed. They should refresh when lookup conditions change, but remain reusable when only source-file selection changes.

- `tsConfigStore.clearCache()` now also resets the prepared state derived from provided `tsConfigPaths`.
  Why: after a cache clear, the store must rebuild that state instead of assuming it is still valid just because the raw property value did not change.

- `canAccessFileSystem` is now part of store invalidation.
  Why: it changes how a store is populated. When it is enabled, the store can be built by walking the project under `baseDir`. When it is disabled, the store can only be built from the explicit `inputFiles` in the request plus simulated parent directories. This check should be read as a defensive cache-safety guard: its purpose is to avoid reusing store state across different filesystem-availability assumptions if those conditions ever share the same long-lived stores.

- `postProcess()` now receives the final analyzable file map from `initFileStores()`.
  Why: post-processing can depend not only on what was discovered during traversal, but also on the exact source-file set that survived filtering for the current analysis. This makes that dependency explicit in the file-store lifecycle.

Overall, the goal of these changes is correct cache reuse: each pre-existing store now stays warm only while the inputs that define its contents are still valid.

File: packages/analysis/src/file-stores/generated-sources.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    if (getProjectFileDiscoveryConfigKey(configuration) !== this.projectFileDiscoveryConfigKey) {
+      this.clearCache();
+      return;
+    }
+
+    for (const filename of fsEvents) {
+      if (this.isRelevantEvent(filename)) {
+        this.clearCache();
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (francois-mora-sonarsource) [reply packages/analysis/src/file-stores/generated-sources.ts:97]: Fixed in ab9d3f3a4 by computing the generated-source watched filenames once per dirtyCachesIfNeeded call and passing the result into isRelevantEvent. Added a regression test that covers multiple fsEvents in one batch.

**Review by vdiez — APPROVED** [review]
This looks good to me now.

I reviewed it against JS-1597 and the current shape matches the ticket well for a foundation PR:
- it introduces the generated-source store and detector contracts without enabling detector-specific behavior yet
- it integrates the new logic into the existing file-store lifecycle instead of adding a parallel ad hoc path
- the cache invalidation split between analyzable-file selection and project-file discovery makes the refresh rules explicit and matches how the stores are populated

What I validated locally:
- `npm ci`
- `npm run bridge:build:fast`
- `packages/analysis/tests/jsts/project-metadata/generated-sources.test.ts`
- `packages/analysis/tests/source-files.test.ts`
- `packages/analysis/tests/tsconfigs.test.ts`
- `packages/analysis/tests/dependency-manifests.test.ts`
- `packages/grpc/tests/analyze-project-normalize.test.ts`

A few reasons why I am comfortable with the current approach:
- the generated-source logic is now wired through the existing stores and configuration keys, so it is not bypassing the analyzer's cache and store infrastructure
- manifest-based inputs flow through `dependencyManifestStore`, and the gRPC normalization path explicitly resets before same-`baseDir` filesystem rediscovery, which keeps the lifecycle coherent
- the rewritten `docs/file-stores.md` is clearer than the older master-era note for the file-store layer, especially around the two cache domains and the new `generated-sources` store

One non-blocking caveat I noticed while checking the docs against the implementation:
- in `canAccessFileSystem=false` mode, the simulated traversal still does not fully mirror real traversal for excluded-directory propagation. Nested files under an excluded directory can remain in `sourceFileStore` because synthetic directories are built leaf-first and ignored-state propagation currently only checks the immediate parent. This looks pre-existing rather than introduced by this PR, so I am not treating it as a blocker here, but it is worth keeping in mind if we rely more on simulated traversal semantics later.
